include/swift/Sema/CSFix.h

```diff
@@ -857,7 +860,8 @@ class ForceOptional final : public ContextualMismatch {
 
   bool diagnose(const Solution &solution, bool asNote = false) const override;
 
-  static ForceOptional *create(ConstraintSystem &cs, Type fromType, Type toType,
+  static ForceOptional *create(ConstraintSystem &cs, Type fromType,
+                               unsigned unwrapCount,
```

Comment:
I think it would be safer to compute this in initializer - we can take optionality of `from` and `to` types, diff them on construction of the fix and add an accessor method to retrieve the number of unwraps so it could be used in `recordFix`. Also note that we'd want to reserve `toType` provided by the caller because it's not necessarily equal to unwrapped from since there could be implicit conversions involved.